tensorflow/lite/experimental/micro/examples/micro_recognition/main.cc

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "mbed.h"
+
+#include "tensorflow/lite/experimental/micro/kernels/all_ops_resolver.h"
+#include "tensorflow/lite/experimental/micro/micro_error_reporter.h"
+#include "tensorflow/lite/experimental/micro/micro_interpreter.h"
+#include "tensorflow/lite/schema/schema_generated.h"
+#include "tensorflow/lite/version.h"
+
+#include "tensorflow/lite/experimental/micro/examples/micro_recognition/disco/display_util.h"
+#include "tensorflow/lite/experimental/micro/examples/micro_recognition/disco/image_util.h"
+#include "tensorflow/lite/experimental/micro/examples/micro_recognition/image_provider.h"
+#include "tensorflow/lite/experimental/micro/examples/micro_recognition/util.h"
+
+#include "tensorflow/lite/experimental/micro/examples/micro_recognition/image_recognition_model.h"
+
+#define NUM_OUT_CH 3
+#define CNN_IMG_SIZE 32
+
+uint8_t camera_buffer[NUM_IN_CH * IN_IMG_WIDTH * IN_IMG_HEIGHT]
+    __attribute__((aligned(4)));
+static const char* labels[] = {"Plane", "Car",  "Bird", "Cat",
+                               "Deer",  "Dog",  "Frog", "Horse",
+                               "Ship",  "Truck"};
+
+int main(int argc, char** argv) {
+  init_lcd();
+  wait_ms(100);
+
+  tflite::MicroErrorReporter micro_error_reporter;
+  tflite::ErrorReporter* error_reporter = &micro_error_reporter;
+
+  if (InitCamera(error_reporter) != kTfLiteOk) {
+    error_reporter->Report("Failed to init camera");
```

Comment:
```
error_reporter->Report("Failed to init camera\n");
```

`\n` is missing.

Reply:
\n isn't neccessary, since DebugLogPrintf() (called by error_reporter->Report()) appends \r\n.
https://github.com/tensorflow/tensorflow/blob/47b46048a97444b332bb7bd36b5097676e2c489d/tensorflow/lite/experimental/micro/micro_error_reporter.cc#L59